File: array/Queue_Static.cs
using System;

public class ArrayMain
{
  public class Queue
  {
    int max, current;
    int[] queue;

    /* Método contrutor da fila recebe o tamanho */
    public Queue(int max)
    {
      /* Inicia a fila com o tamanho determinado */
      this.queue = new int[max];

      this.max = max;
      this.current = -1;
    }

    /* Verifica se a fila não está cheia e adiciona o elemento.  */
    public void Enqueue(int element)
    {
      if(this.current + 1 == this.max) return;

      this.queue[++this.current] = element;
    }

    /* Verifica se a fila não está vazia e remove um elemento. */
    public int Dequeue()
    {
      if(this.current == -1) throw new Exception("Empty");

      int element = this.queue[0];
      for(int i = 0; i < this.current;i++)
      {
        this.queue[i] = this.queue[i+1];
      }

      this.queue[this.current--] = 0;

      return element;
    }
  }

  public static void Main()
  {
    Queue queue = new Queue(5);
    queue.Enqueue(1);
    queue.Enqueue(2);
    queue.Enqueue(3);
    queue.Enqueue(4);
    queue.Enqueue(5);

    Console.WriteLine("Dequeue > {0}", queue.Dequeue());
    Console.WriteLine("Dequeue > {0}", queue.Dequeue());
    Console.WriteLine("Dequeue > {0}", queue.Dequeue());
    Console.WriteLine("Dequeue > {0}", queue.Dequeue());
    Console.WriteLine("Dequeue > {0}", queue.Dequeue());

    /* Gera Exception por não ter mais elementos na fila. */
    //Console.WriteLine("Dequeue > {0}", queue.Dequeue());
  }
}

File: array/Queue_Static.cpp
#include <bits/stdc++.h>

using namespace std;

class Queue {

  int *queue, max, current;
  public:
    Queue(int max)
    {
      this->queue = new int[max];

      this->max = max;
      this->current = -1;
    }

    void enqueue(int element){
      if( this->current + 1 == this->max ) return;

      this->queue[++this->current] = element;
    }

    int dequeue(){
      if( this->current == -1 ) throw "Empty";

      int element = this->queue[0];
      for(int i = 0; i < this->current; i++){
        queue[i] = queue[i+1];
      }

      this->queue[current--] = 0;

      return element;
    }
};

int main()
{
  Queue queue(5);
  queue.enqueue(1);
  queue.enqueue(2);
  queue.enqueue(3);
  queue.enqueue(4);
  queue.enqueue(5);

  printf("Dequeue > %d\n", queue.dequeue());
  printf("Dequeue > %d\n", queue.dequeue());
  printf("Dequeue > %d\n", queue.dequeue());
  printf("Dequeue > %d\n", queue.dequeue());
  printf("Dequeue > %d\n", queue.dequeue());

  /* Gera Exception por não ter mais elementos na fila */
  //printf("Dequeue > %d\n", queue.dequeue());

  return 0;
}
